### PR TITLE
fix: handle edge case when from_block is greater than latest_block_number

### PR DESCRIPTION
**Pull request by lambda-0x**
fix: #2193

### CODE REVIEW

File: crates/rpc/src/method/get_events.rs [hunk 1]
```diff
@@ -155,6 +155,39 @@ pub async fn get_events(
...
+
+            let pending = context
+                .pending_data
+                .get(&transaction)
+                .context("Querying pending data")?;
+
+            // `from_block` is the pending block's number
+            if from_block == pending_block_number {
+                return get_pending_events(&request, &pending, continuation_token);
+            }
```
Comment (lambda-0x) [inline crates/rpc/src/method/get_events.rs:188]: after thinking about it a bit more i think it would be more confusing than helpful so will remove this for now, happy to year your thoughts on this.

**Review by kkovaacs — COMMENTED** [review]
We should also add a test case for this edge case. This seemed to work for me in the `tests::pending` module:

```
        #[tokio::test]
        async fn from_block_past_pending() {
            let context = RpcContext::for_tests_with_pending().await;

            let input = GetEventsInput {
                filter: EventFilter {
                    from_block: Some(BlockId::Number(BlockNumber::new_or_panic(4))),
                    to_block: Some(BlockId::Pending),
                    chunk_size: 100,
                    ..Default::default()
                },
            };
            let result = get_events(context, input).await.unwrap();
            assert!(result.events.is_empty());
        }
```

File: crates/rpc/src/method/get_events.rs [hunk 2]
```diff
@@ -155,6 +156,29 @@ pub async fn get_events(
...
+            let pending_block_number = *latest_block_number + 1;
+
+            // `from_block` is larger than pending block's number
+            if from_block > pending_block_number {
+                return Ok(types::GetEventsResult {
+                    events: Vec::new(),
+                    continuation_token: None,
+                });
+            }
+        }
```
[the remark this answers is not in the document]
Comment (lambda-0x) [reply crates/rpc/src/method/get_events.rs:180]: oh yes, this looks much cleaner!

**Review by kkovaacs — APPROVED** [review]
LGTM, thank you!
